### PR TITLE
bring annotations to API

### PR DESCRIPTION
**Pull request by nicosomb**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | WIP
| Documentation | no
| Translation   | no
| Fixed tickets | https://github.com/wallabag/wallabag/issues/1800
| License       | MIT

### CODE REVIEW

**Comment by tcitworld** [comment]
Why ?

**Comment by nicosomb** [comment]
@tcitworld Do you need help for this PR?

**Comment by nicosomb** [comment]
@tcitworld what is the TODO list on this PR?

**Comment by tcitworld** [comment]
This needs more work : delete the AnnotationBundle Controllers and handle them in APIBundle. Not sure the authentication checks support direct calls without OAuth though.

**Comment by jlnostr** [comment]
So, annotations won't come to the API in 2.1.0, did I understand this right?

**Comment by j0k3r** [comment]
Yep sorry @jlnostr we won't have enough free time to work on that before releasing 2.1.0. :confused:

**Comment by jlnostr** [comment]
Alright, no problem :smile:

**Comment by tcitworld** [comment]
> Not sure the authentication checks support direct calls without OAuth though.

It works, but then we loose the whole point of REST. Won't do it.
`/annotation` is for the browser and `/api/annotation` will be for the api.

**Comment by nicosomb** [comment]
> `/annotation` is for the browser and `/api/annotation` will be for the api.

Is that a problem?

**Comment by tcitworld** [comment]
Not at all.

File: src/Wallabag/AnnotationBundle/Controller/WallabagAnnotationController.php [hunk 1]
```diff
@@ -15,42 +14,35 @@ class WallabagAnnotationController extends FOSRestController
...
         $annotationRows = $this
-                ->getDoctrine()
-                ->getRepository('WallabagAnnotationBundle:Annotation')
-                ->findAnnotationsByPageId($entry->getId(), $this->getUser()->getId());
+            ->getDoctrine()
+            ->getRepository('WallabagAnnotationBundle:Annotation')
+            ->findAnnotationsByPageId($entry->getId(), $this->getUser()->getId());
         $total = count($annotationRows);
-        $annotations = ['total' => $total, 'rows' => $annotationRows];
+        $annotations = array('total' => $total, 'rows' => $annotationRows);
```
Comment (j0k3r) [inline src/Wallabag/AnnotationBundle/Controller/WallabagAnnotationController.php]: Nope, please use short syntax

File: src/Wallabag/ApiBundle/Controller/WallabagRestController.php [hunk 2]
```diff
@@ -470,6 +472,115 @@ public function deleteTagAction(Tag $tag)
...
+    public function getAnnotationsAction(Entry $entry)
+    {
+        $this->validateAuthentication();
+
+        $response = $this->forward('WallabagApiBundle:WallabagRest:getAnnotations',
+            [
+                'entry' => $entry,
+            ]);
+
+        return $response;
```
Comment (j0k3r) [inline src/Wallabag/ApiBundle/Controller/WallabagRestController.php]: You can put sth more simplier:

```php
return $this->forward('WallabagApiBundle:WallabagRest:getAnnotations', [
    'entry' => $entry,
];
```

Same for other actions below

**Comment by nicosomb** [comment]
OK, tests are KO, I know why. Will fix them.

**Comment by tcitworld** [comment]
This is postgresql !

**Pull request merged**